persistence/pre_persistence/account.go

```diff
@@ -0,0 +1,377 @@
+package pre_persistence
+
+import (
+	"bytes"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"math/big"
+)
+
+const (
+	ServiceNodeStakePoolName = "SERVICE_NODE_STAKE_POOL"
+	AppStakePoolName         = "APP_STAKE_POOL"
+	ValidatorStakePoolName   = "VALIDATOR_STAKE_POOL"
+	FishermanStakePoolName   = "FISHERMAN_STAKE_POOL"
+	DAOPoolName              = "DAO_POOL"
+	FeePoolName              = "FEE_POOL"
+)
+
+func (m *PrePersistenceContext) AddPoolAmount(name string, amount string) error {
+	cdc := Cdc()
+	p := Pool{}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	val, err := db.Get(key)
+	if err != nil {
+		return err
+	}
+	err = cdc.Unmarshal(val, &p)
+	if err != nil {
+		return err
+	}
+	s, err := StringToBigInt(p.Account.Amount)
+	if err != nil {
+		return err
+	}
+	s2, err := StringToBigInt(amount)
+	if err != nil {
+		return err
+	}
+	s.Add(s, s2)
+	p.Account.Amount = BigIntToString(s)
+	bz, err := cdc.Marshal(&p)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) SubtractPoolAmount(name string, amount string) error {
+	cdc := Cdc()
+	p := Pool{}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	val, err := db.Get(key)
+	if err != nil {
+		return err
+	}
+	err = cdc.Unmarshal(val, &p)
+	if err != nil {
+		return err
+	}
+	s, err := StringToBigInt(p.Account.Amount)
+	if err != nil {
+		return err
+	}
+	s2, err := StringToBigInt(amount)
+	if err != nil {
+		return err
+	}
+	s.Sub(s, s2)
```

Comment:
1. Do we want to add a check to make sure that s is greater than s2 before subtracting?
1.1 If `yes`, this might warrant adding a simple unit test since it seems to have fallen through the cracks.
1.2 If `not`, ignore this comment altogether.

Reply:
This is checked at the Utility level (an unfortunate byproduct of piece mealing the PR), but if we want to reject it at the persistence layer we could as well

Reply:
Got it, sgtm.